pkg/loop/internal/core/services/capability/capabilities.go

```diff
@@ -270,6 +292,17 @@ func (t *triggerExecutableClient) RegisterTrigger(ctx context.Context, req capab
 		return nil, fmt.Errorf("error registering trigger: %w", err)
 	}
 
+	// In order to ensure the registration is successful, we need to wait for the first message from the server.
+	// This will be an ack or error message. If the error is not nil, we return an error.
+	ackMsg, err := responseStream.Recv()
```

Comment:
Should we add a timeout on how long we will wait for this?

Reply:
We pass in the context when calling RegisterTrigger above; that will scope the stream as a whole